state-chain/pallets/cf-elections/src/electoral_systems/block_witnesser/primitives.rs

```diff
@@ -383,6 +392,10 @@ impl<N: Step + Ord> CompactHeightTracker<N> {
 			range_difference(r, &(range.start().clone()..N::forward(range.end().clone(), 1)))
 		}
 	}
+
+	fn get_all_heights(&self) -> BTreeSet<N> {
```

Comment:
We could just return an iterator here and let the consumer decide what it wants to collect into - especially if the only use so far is turning it into an iterator

Reply:
Unfortunately this doesn't really improve things since the way we use this is that we want an iterator over `&N` references (we let this interact with the `&N`s that are the keys of the other ongoing and queued BTreeMaps). Here we can't return an `&N` because the ranges over that we iterate don't actually store all the `N`s, and so can only give us an iterator over `N`.

The alternative would be to change the other iterators to be over `N` as well, but that would mean more copying, in particular since we expect the safe block heights to be at most very sparsely populated.